packages/std/src/exports.rs

```diff
@@ -21,6 +21,10 @@ use crate::serde::{from_slice, to_vec};
 use crate::types::Env;
 use crate::{Deps, DepsMut, MessageInfo};
 
+#[cfg(feature = "iterator")]
```

Comment:
Ah nice, so no more logic added to the VM, we just make the signalling explicit, so a contract that requires it would fail on upload on eg. Secret rather than make annoying errors later on (eg in queries).

Nice idea.